### ISSUE TITLE
Can’t use box with version contains minor number leading zeroes

### ISSUE THREAD

**Issue by yukihiko-shinoda** [themalkolm/packer-builder-vagrant#11]
Hello, thank you for your useful plugin development.
In specification of Vagrant,
it allows version contains minor number leading zeroes.
https://app.vagrantup.com/bento/boxes/centos-6.9

However, semver seams to be not allow version contains minor number leading zeroes.
https://github.com/blang/semver/blob/557ee991fbcf208593ae816b71e43e0c6485f21f/semver.go#L255

So, we can’t get succeed packer with packer-plugin-vagrant.

```
$ vagrant box list
bento/centos-6.9                           (virtualbox, 2.3.5)
bento/centos-6.9                           (virtualbox, 2.3.8)
bento/centos-6.9                           (virtualbox, 201802.02.0)

$ packer build packer_template.json
vagrant output will be in this color.

    vagrant: (vagrant) Builder source_path ...
==> vagrant: Minor number must not contain leading zeroes "02"
Build 'vagrant' errored: Minor number must not contain leading zeroes "02"

==> Some builds didn't complete successfully and had errors:
--> vagrant: Minor number must not contain leading zeroes "02"

==> Builds finished but no artifacts were created.
```

Could you allow version contains minor number leading zeroes?
(Sorry, I can’t help golang development...)

**Comment by themalkolm** [comment]
What version of this plugin do you use?

**Comment by yukihiko-shinoda** [comment]
Last time, I used v2018.01.31.
Today, even I used v2018.03.01, I got same result.

Packer: 1.1.3
Vagrant: 2.0.3
VirtualBox: 5.2.6

**Comment by themalkolm** [comment]
Gotcha.

**Comment by themalkolm** [comment]
```
$ cd test/virtualbox
$ ./test-issue-11.sh
...
    vagrant: (vagrant) Builder source_path ...
    vagrant: (vagrant) Downloading box: bento/centos-6.9 (virtualbox, 201802.02.0)
    vagrant: (vagrant) box: Loading metadata for box 'bento/centos-6.9'
    vagrant: (vagrant) box: URL: https://vagrantcloud.com/bento/centos-6.9
    vagrant: (vagrant) box: Adding box 'bento/centos-6.9' (v201802.02.0) for provider: virtualbox
    vagrant: (vagrant) box: Downloading: https://vagrantcloud.com/bento/boxes/centos-6.9/versions/201802.02.0/providers/virtualbox.box
==> box: Successfully added box 'bento/centos-6.9' (v201802.02.0) for 'virtualbox'!
    vagrant: (vagrant) Downloaded box: bento/centos-6.9 (virtualbox, 201802.02.0)
    vagrant: (vagrant) Builder source_path: /Users/malkolm/.vagrant.d/boxes/bento-VAGRANTSLASH-centos-6.9/201802.02.0/virtualbox/box.ovf
    vagrant: (vagrant) Builder type ...
    vagrant: (vagrant) Builder type: virtualbox-ovf
...
```

Changed from semver to go-version. Seems to work now fine.

**Comment by themalkolm** [comment]
Fixed in v2018.04.15 https://github.com/themalkolm/packer-builder-vagrant/releases/tag/v2018.04.15. Note that I bumped required packer version to 1.2.2.

Tested fix on:

vagrant 2.0.3
packer 1.2.2
VirtualBox 5.2.8
